#18 - Implemented basic django admin view for DBTaskResult

jurecuhalev opened the pull request:
Changes:
- Configures basic settings to include django admin (+auth, static files)
- Add basic django admin view
- Django admins are configured to be read-only

Screenshots:

<img width="1155" alt="Screenshot 2024-06-08 at 12 42 08" src="https://github.com/RealOrangeOne/django-tasks/assets/13806/303b0f92-7b0a-41bf-9a33-9a9ca8973a2b">

<img width="1479" alt="Screenshot 2024-06-08 at 12 42 16" src="https://github.com/RealOrangeOne/django-tasks/assets/13806/abc4a931-39ac-4a03-8b4d-93c5229365ed">

Ref #18

RealOrangeOne commented on tests/urls.py:
```diff
@@ -1,7 +1,10 @@
 
 from . import views
 
+admin.autodiscover()
```
Issue: 

> Typically you won’t need to call this function directly as [AdminConfig](https://docs.djangoproject.com/en/5.0/ref/contrib/admin/#django.contrib.admin.apps.AdminConfig) calls it when Django starts.

https://docs.djangoproject.com/en/5.0/ref/contrib/admin/#django.contrib.admin.autodiscover

jurecuhalev replied:
Updated PR to remove it.

RealOrangeOne commented on django_tasks/backends/database/admin.py:
```diff
@@ -0,0 +1,15 @@
+    list_display = ("id", "task_path", "status", "priority", "queue_name")
+    list_filter = ("status", "priority", "queue_name")
+
+    def has_add_permission(self, request, obj=None):
```
Issue: I'd suggest also adding `has_delete_permission` and `has_change_permission`:

```python
    def has_delete_permission(self, request, obj=None):
        return False

    def has_change_permission(self, request, obj=None):
        return False
```

This removes a few buttons, which cleans up the UI a little.

RealOrangeOne commented on tests/settings.py:
```diff
@@ -6,30 +6,48 @@
 django_stubs_ext.monkeypatch()
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+DEBUG = True
```
Question: What is `DEBUG` needed for? This changes the semantics of the tests a little, so we shouldn't do it unless we **have** to.

Perhaps this too can be moved to a separate settings file along with the `TASKS` setting to be used when someone is running the test app directly, vs using it for the test suite.

jurecuhalev replied:
We need debug for static files in django

RealOrangeOne commented on django_tasks/backends/database/admin.py:
```diff
@@ -0,0 +1,15 @@
+
+@admin.register(DBTaskResult)
+class DBTaskResultAdmin(admin.ModelAdmin):
+    list_display = ("id", "task_path", "status", "priority", "queue_name")
```
Suggestion: It might be easier to read to use the task's `.name` attribute, as it'll be more human-readable than the full path (not to mention much shorter)

```suggestion
    list_display = ("id", "task_path", "status", "priority", "queue_name")
```

RealOrangeOne commented on django_tasks/backends/database/admin.py:
```diff
@@ -0,0 +1,22 @@
+
+
+@admin.register(DBTaskResult)
+class DBTaskResultAdmin(admin.ModelAdmin):
```
Suggestion: Now that we're exposing their names, it might be nice to set nicer names for the module and the `verbose_name` for the model itself, so the admin interface looks a little nicer.

jurecuhalev commented:
Changes in this push:
- Added verbose name (to DBTaskResult and App itself)
- Updated settings to not turn on DEBUG or default Task backend if tests are running
- Improved admin display and removed delete/change actions

RealOrangeOne commented on tests/settings.py:
```diff
@@ -1,32 +1,52 @@
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
+if sys.argv[0] != "test":
```
Nit: This check should probably be at the bottom, so it supersedes the other settings.

jurecuhalev replied:
Done
